File: 011 Container With Most Water/container_with_most_water.cs
public class Solution {	// Author: Xinyu Chen
    public int MaxArea(int[] height) {
        int volume = 0;
        int lo = 0, hi = height.Length - 1;
        while (lo < hi) {
            int h = Math.Min(height[lo], height[hi]);
            volume = Math.Max(volume, (hi - lo) * h);
            while (height[lo] <= h && lo < hi) lo++;
            while (height[hi] <= h && lo < hi) hi--;
        }
        return volume;
    }
}

File: 011 Container With Most Water/container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0, j = height.size() - 1;
        int largest = 0;
        int curr;
        while (i < j) {
            curr = min(height[i], height[j]) * (j - i);
            largest = max(largest, curr);
            if (height[i] < height[j]) i++;
            else j--;
        }
        return largest;
    }
}; // Author: XC